### PR TITLE
PROJECT: ASSOCIATIONS

### PR DESCRIPTION
**Pull request by macnick**
We have made a cute little project. Please review our code.

### CODE REVIEW

File: app/models/event.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+class Event < ApplicationRecord
+  scope :upcoming, ->(n = Date.today.to_s) { where('date > ?', n) }
```
Comment (samgaco) [inline app/models/event.rb]: - [ ] you should add and = sign so the event that tales place _today_ appears somewhere as well

File: app/models/event.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  scope :past, ->(n = Date.today.to_s) { where('date < ?', n) }
+  has_many :invitations
+  has_many :users, through: :invitations
+
```
Comment (samgaco) [inline app/models/event.rb]: - [ ] the event model is created by one user only, where is that represented? The event model should have one [creator](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_assoc_requirement.png).

File: db/migrate/20191126141806_create_events.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+class CreateEvents < ActiveRecord::Migration[5.2]
+  def change
+    create_table :events do |t|
+      t.string :title
+      t.string :location
+      t.integer :user_id
+      t.datetime :date
```
Comment (samgaco) [inline db/migrate/20191126141806_create_events.rb]: - [ ] We should have a foreign key to represent the [creator](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_assoc_requirement.png) of the event

File: app/controllers/events_controller.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    @upcoming = Event.upcoming
+    @past = Event.past
+  end
+
+  def new
+    @event = current_user.events.build
+  end
+
+  def create
+    @event = Event.new(event_params)
```
Comment (samgaco) [inline app/controllers/events_controller.rb]: - [ ] You should be using the method [build](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/build_method_requirement.png) for the create action

File: app/controllers/events_controller.rb [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    else
+      flash[:danger] = 'User was not created'
+      render 'new'
+    end
+  end
+
+  def show
+    @event = Event.find(params[:id])
+    @creator = User.find(@event.user_id)
+  end
```
Comment (samgaco) [inline app/controllers/events_controller.rb]: - [ ] You should use your created associations to list the created events by the current user, you can also use the current_user helper

- [ ] we should also have the upcoming and past events instantiated in [here](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png)

File: app/views/users/show.html.erb [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+			<%= will_paginate @events %>
+		<% else %>
+			<h3>No Events Found</h3>
+		<% end %>
+	</div>
+
+	<div class="col-md-6">
+		<% if @created.any? %>
+			<h3 class="text-center">Created events</h3>
+			<% @created.select {|e| e.user_id == @user.id}.each do |event| %>
```
Comment (samgaco) [inline app/views/users/show.html.erb]: - [ ] You shouldn't have any kind of logic on your views, the created events by a given user should be accessible with your created associations in the models and instances in the controller actions.

File: app/models/event.rb (hunk 1 above)
Comment (macnick) [reply app/models/event.rb]: Sorry can not understand what you mean here.

File: app/controllers/events_controller.rb (hunk 4 above)
Comment (macnick) [reply app/controllers/events_controller.rb]: This one is fixed

File: db/migrate/20191205151538_add_creator_to_events.rb [hunk 7]
```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+class AddCreatorToEvents < ActiveRecord::Migration[5.2]
+  def change
+    add_column :events, :creator_id, :integer
+  end
+  add_index :events, :creator_id
```
Comment (samgaco) [inline db/migrate/20191205151538_add_creator_to_events.rb]: :+1:

**Comment by macnick** [comment]
Thank you so much!!

**Comment by addod19** [comment]
Merging Association Project to the development branch

**Pull request merged**